durastore/src/main/java/org/duracloud/durastore/rest/AuxRest.java

```diff
@@ -86,4 +101,19 @@ public Response getCookies(@QueryParam("token") String token) {
         }
     }
 
+    private S3StorageProvider getS3StorageProvider() {
+        if (!storageAccountManager.isInitialized()) {
+            throw new DuraCloudRuntimeException("storageAccountManager is not initialized!!!");
+        }
+        StorageAccount account = storageAccountManager.getPrimaryStorageAccount();
+        StorageProviderType type = account.getType();
+        if (!type.equals(StorageProviderType.AMAZON_S3)) {
```

Comment:
This type check should be removed. There are storage provider types build on top of the AMAZON_S3 implementation which would have a different type, but would still work just as well for streaming. For example, using the CHRONOPOLIS provider but where some spaces are not used for snapshots (left in S3) could support this. Also, the 4Science-proposed AMAZON_ONE type would fit this category.